src/backend/distributed/commands/statistics.c

```diff
@@ -127,6 +132,55 @@ CreateStatisticsStmtObjectAddress(Node *node, bool missingOk)
 }
 
 
+/*
+ * PreprocessDropStatisticsStmt is called during the planning phase for
+ * DROP STATISTICS.
+ */
+List *
+PreprocessDropStatisticsStmt(Node *node, const char *queryString)
+{
+	DropStmt *dropStatisticsStmt = castNode(DropStmt, node);
+	Assert(dropStatisticsStmt->removeType == OBJECT_STATISTIC_EXT);
+
+	if (!ShouldPropagate())
+	{
+		return NIL;
+	}
+
+	QualifyTreeNode((Node *) dropStatisticsStmt);
+
+	Oid relationId;
+	List *ddlJobs = NIL;
+	List *objectNameList = NULL;
+	foreach_ptr(objectNameList, dropStatisticsStmt->objects)
+	{
+		relationId = GetRelIdByStatName(objectNameList, dropStatisticsStmt->missing_ok);
+
+		if (!OidIsValid(relationId) || !IsCitusTable(relationId))
+		{
+			continue;
+		}
+
+		EnsureCoordinator();
```

Comment:
We also call EnsureCoordinator() in ExecuteDistributedDDLJob, so it's unnecessary here.

If a day comes when we want to allow DDL from workers, it will be easier if we have fewer calls to remove.